feathr_project/feathr/_synapse_submission.py

```diff
@@ -224,20 +225,19 @@ def create_spark_batch_job(self, job_name, main_file, class_name=None,
         executor_cores = self.EXECUTOR_SIZE[self._executor_size]['Cores']
         executor_memory = self.EXECUTOR_SIZE[self._executor_size]['Memory']
 
-        # Adding spaces between brackets. This is to workaround this known YARN issue (when running Spark on YARN):
-        # https://issues.apache.org/jira/browse/SPARK-17814?focusedCommentId=15567964&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15567964
-        # updated_arguments = []
-        # for elem in arguments:
-        #     if type(elem) == str:
-        #         updated_arguments.append(elem.replace("}", " }"))
-        #     else:
-        #         updated_arguments.append(elem)
+        # need to put the jar in as dependencies for pyspark job
+        jars = jars + [main_file]
+
+        # If file=ain_file, then it's using only Scala Spark
+        # If file=python_files[0], then it's using Pyspark
+        spark_execution_file = python_files[0] if python_files else main_file
 
         spark_batch_job_options = SparkBatchJobOptions(
             tags=tags,
             name=job_name,
-            file=main_file,
+            file=spark_execution_file,
             class_name=class_name,
+            python_files=python_files[1:],
```

Comment:
Check python_files before call [1:] ?

Reply:
it's guranteed to be non-None. So if it's less than 2 elements, it will just always be [] which is ok.